extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java

```diff
@@ -831,18 +831,20 @@ private boolean isUsingVertxRoute(IndexView index) {
         return false;
     }
 
-    private OpenAPI generateStaticModel(SmallRyeOpenApiConfig openApiConfig, List<Pattern> ignorePatterns, Path target)
+    private OpenAPI generateStaticModel(SmallRyeOpenApiConfig smallRyeOpenApiConfig, List<Pattern> ignorePatterns, Path target)
             throws IOException {
-        if (openApiConfig.ignoreStaticDocument) {
+        Config config = ConfigProvider.getConfig();
+        OpenApiConfig openApiConfig = OpenApiConfigImpl.fromConfig(config);
```

Comment:
It might be better to obtain the `Config` and `OpenApiConfig` in the `build` method that calls `generateStaticModel`, and pass that here and also to `generateAnnotationModel`.